### DISCUSSION TITLE
ALTER TABLE Tab ADD TimeIns TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL

### DISCUSSION THREAD

**Discussion by rene-weiss** [pawelsalawa/sqlitestudio#3946] [Q&A]
### Details
Change Table ALTER  will not work... Error:
--> Cannot add a column with non-constant default

### Steps to reproduce
this works...:
CREATE TABLE my_table (
    id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,
    name VARCHAR(64),
    sqltime TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL
)

This not:
ALTER TABLE tabTmx ADD TimeIns TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL

### Operating system
Win10
### SQLiteStudio version
3.2.1

**Comment by pawelsalawa** [comment]
This is how SQLite3 database works and it's intentional. See point `4. ALTER TABLE ADD COLUMN` at the page: https://sqlite.org/lang_altertable.html

**Comment by rene-weiss** [comment]
Thanks
